source/tailwind-provider.tsx

```diff
@@ -1,20 +1,22 @@
-import * as React from 'react';
-import {useColorScheme} from 'react-native';
 import {
-	useDimensions,
 	useAccessibilityInfo,
-	useDeviceOrientation
+	useDeviceOrientation,
+	useDimensions
 } from '@react-native-community/hooks';
-import TailwindContext from './tailwind-context';
+import * as React from 'react';
+import {useColorScheme, ColorSchemeName} from 'react-native';
+
 import create from './create';
+import TailwindContext from './tailwind-context';
 import {Utilities} from './types';
 
 interface Props {
 	utilities: Utilities;
+	theme?: ColorSchemeName;
 }
 
-const TailwindProvider: React.FC<Props> = ({utilities, children}) => {
-	const colorScheme = useColorScheme() ?? 'light';
+const TailwindProvider: React.FC<Props> = ({utilities, theme, children}) => {
+	const colorScheme = theme ?? useColorScheme() ?? 'light';
```

Comment:
shouldn't conditionally call hooks. Add this logic in the  useMemo instead  

```
	const colorScheme = useColorScheme() ?? 'light';
        ...
       	const tailwind = React.useMemo(() => {
		return create(utilities, {
			colorScheme: theme ?? colorScheme,
			width,
			height,
			reduceMotion: Boolean(reduceMotion),
			orientation
		});
	}, [utilities, colorScheme, width, height, reduceMotion, orientation, theme]);

Reply:
Hi @HeleneGrini, thanks for reviewing it. What is the difference? 
In case theme or colorScheme has changed, tailwind will change as well. Maybe I'm missing something, but could you please extend your thoughts a little? Thanks!

Reply:
Thanks for making the PR! 😄 

It's one of the rules of hooks -- they shouldn't be called in loops, conditions, or  functions (other than hooks or komponents) https://reactjs.org/docs/hooks-rules.html 

It can result in the order/number of hook being different on different renders, which will result in an error as React stores the value of hooks in an ordered manner. If the order changes, the hooks will not me matched to their appropriate value.   



An example here would be that you have a state in your app storing the `theme`. Let's say your state is `undefined` unless the user has manually set a theme in you app. This will then default to the system colorScheme in the TailwindProvider. `useColorScheme` gets called, then `useMemo` gets called. Later your user sets the theme in your app, the props to the TailwindProvider change causing a rerender. On this render theme props is set and `useColorScheme` is _not_ called. `useMemo` gets called as the first hook, and will the be matched to first stored hook-value in react, but that was expected to be `useColorScheme` as it was the previous render. Booom, error 

I would suggest having a look at this explanation https://reactjs.org/docs/hooks-rules.html#explanation

Reply:
Hi @HeleneGrini, interesting topic here. Maybe I misunderstood everything, but I am changing a variable at runtime. Regarding your example, anytime either `colorScheme` and changed (either via `useColorScheme` or `theme`), tailwind with be changed as well. 

Let's say:
```
useColorScheme() returns 'light';
theme = undefined;

`create` is called and tailwind will return light theme styles;

user change theme to `light` => `create` will not be called, because colorScheme is equal to light, therefore there's no need to update it.
user change theme to 'dark' => `create` is called, because `colorScheme` has changed, prev light, current dark;
```

or 

```
useColorScheme() returns undefined;
theme = undefined;
colorScheme will be `light` (the fallback)

`create` is called and tailwind will return light theme styles;

and now follows the flow presented from above.
```

Maybe I don't understand this concept enough and I can be completely wrong.

`colorScheme` its a variable + a dependency of useMemo which can take a value of `theme` or `useColorScheme` or `light`, we don't call a hook conditionally here.

Reply:
The issue is not with the tailwind variable, but rather the colorScheme variabel 

```
const colorScheme = theme ?? useColorScheme() ?? 'light';
```
 if `theme` from props is `undefined`  `useColorScheme` is called, if `theme` from props is set `useColorScheme`  is _not_ called hence you are conditionally calling the hook. 
 
 The value of `theme` from props may change from `undefined` to `'light' | 'dark'` or vice versa, which would result in an error

Reply:
Oh dayum, thanks for the clarification. I am going tonpush an update. 🤗

Reply:
Done, please take another look. Thanks!